presentation/src/main/java/com/pomonyang/mohanyang/presentation/designsystem/toast/MnToastSnackbarHost.kt

```diff
@@ -0,0 +1,141 @@
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SnackbarDuration
+import androidx.compose.material3.SnackbarHost
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.SnackbarResult
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.graphics.Color
+import com.mohanyang.presentation.R
+import com.pomonyang.mohanyang.presentation.designsystem.icon.MnMediumIcon
+import com.pomonyang.mohanyang.presentation.designsystem.token.MnColor
+import com.pomonyang.mohanyang.presentation.designsystem.token.MnRadius
+import com.pomonyang.mohanyang.presentation.designsystem.token.MnSpacing
+import com.pomonyang.mohanyang.presentation.theme.MnTheme
+import com.pomonyang.mohanyang.presentation.util.ThemePreviews
+import com.pomonyang.mohanyang.presentation.util.noRippleClickable
+import kotlinx.coroutines.launch
+import timber.log.Timber
+
+@Composable
+fun MnToastSnackbarHost(
+    hostState: SnackbarHostState,
+    modifier: Modifier = Modifier,
+    leadingIconResourceId: Int? = null,
+    leadingIconColor: Color = MnTheme.iconColorScheme.inverse
+) {
+    SnackbarHost(
+        hostState = hostState,
+        modifier = modifier.alpha(0.9f)
+    ) { data ->
+        Card(
+            shape = RoundedCornerShape(MnRadius.small),
+            colors = CardDefaults.cardColors(
+                containerColor = MnTheme.backgroundColorScheme.inverse,
+                contentColor = MnColor.White
+            )
+        ) {
+            Row(
+                modifier = Modifier.padding(MnSpacing.large),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                if (leadingIconResourceId != null) {
+                    MnMediumIcon(resourceId = leadingIconResourceId, tint = leadingIconColor)
+                    Spacer(modifier = Modifier.padding(end = MnSpacing.small))
+                }
+
+                Column(modifier = Modifier.weight(1f)) {
+                    Text(
+                        text = data.visuals.message,
+                        style = MnTheme.typography.subBodySemiBold
+                    )
+                }
+
+                if (data.visuals.actionLabel != null) {
+                    Spacer(modifier = Modifier.padding(start = MnSpacing.small))
+                    Text(
+                        modifier = Modifier.noRippleClickable {
+                            data.performAction()
+                        },
+                        text = data.visuals.actionLabel!!,
+                        style = MnTheme.typography.subBodySemiBold
+                    )
```

Comment:
아마 `data.visuals.actionLabel` 이 스마트 캐스팅이 안되어서 `!!`를 쓴거 같은데 아래와 같이 사용하거나 `?.let` 을 사용해보는건 어떨까?

```suggestion
               val actionLabel = data.visuals.actionLabel
                if (actionLabel != null) {
                    Spacer(modifier = Modifier.padding(start = MnSpacing.small))

                    Text(
                        modifier = Modifier.noRippleClickable {
                            data.performAction()
                        },
                        text = actionLabel,
                        style = MnTheme.typography.subBodySemiBold
                    )
                }
```

Reply:
반영해서 올릴게용